TimeTracking/TimeTracking.php

```diff
@@ -20,239 +20,266 @@
-				$t_record['bugnote_id'] = $p_bugnote_id;
-				$t_record['bug_id'] = $p_bug_id;
-				TimeTracking\create_record( $t_record );
-			}
-		}
-	}
-
-	function ev_view_bug_details( $p_event, $p_bug_id ) {
-		if( TimeTracking\user_can_view_bug_id( $p_bug_id ) ) {
-			$t_records = TimeTracking\get_records_for_bug( $p_bug_id );
-			if( $t_records ) {
-				TimeTracking\print_bug_details_row( $p_bug_id );
-			}
-		}
-	}
-
-	function ev_layout_content_begin( $p_event ) {
-		if( TimeTracking\stopwatch_enabled() && TimeTracking\stopwatch_exists() ) {
-			TimeTracking\print_stopwatch_header_control();
-		}
-	}
-
-	function ev_view_bug( $p_event, $p_bug_id ) {
-		if( TimeTracking\user_can_view_bug_id( $p_bug_id ) ) {
-			TimeTracking\print_bug_timetracking_section( $p_bug_id );
-		}
-	}
-
-	function schema() {
-		$schema[0] =
-			array( 'CreateTableSQL', array( plugin_table( 'data' ), "
-				id                 I       NOTNULL UNSIGNED AUTOINCREMENT PRIMARY,
-				bug_id             I       DEFAULT NULL UNSIGNED,
-				user               I       DEFAULT NULL UNSIGNED,
-				expenditure_date   T       DEFAULT NULL,
-				hours              F(15,3) DEFAULT NULL,
-				timestamp          T       DEFAULT NULL,
-				category           C(255)  DEFAULT NULL,
-				info               C(255)  DEFAULT NULL
-				" )
-		);
-
-		$schema[1] =
-			array(
-				'AddColumnSQL',
-				array( plugin_table( 'data' ),
-					"	user_id			I       UNSIGNED,
-						date_created	I		UNSIGNED NOTNULL DEFAULT '1',
-						time_count		I		UNSIGNED NOTNULL DEFAULT '0',
-						time_exp_date	I		UNSIGNED NOTNULL DEFAULT '1',
-						bugnote_id		I		UNSIGNED"
-					)
-				);
-
-		$schema[2] = array( 'UpdateFunction', 'date_migrate', array( plugin_table( 'data' ), 'id', 'expenditure_date', 'time_exp_date' ) );
-		$schema[3] = array( 'UpdateFunction', 'date_migrate', array( plugin_table( 'data' ), 'id', 'timestamp', 'date_created' ) );
-		$schema[4] = array( 'UpdateFunction', 'timetracking_update_hours', array() );
-		$schema[5] = array( 'UpdateFunction', 'timetracking_update_user_id', array() );
-
-		$schema[6] = array( 'DropColumnSQL', array( plugin_table( 'data' ), 'user' ) );
-		$schema[7] = array( 'DropColumnSQL', array( plugin_table( 'data' ), 'expenditure_date' ) );
-		$schema[8] = array( 'DropColumnSQL', array( plugin_table( 'data' ), 'timestamp' ) );
-		$schema[9] = array( 'DropColumnSQL', array( plugin_table( 'data' ), 'hours' ) );
-
-		return $schema;
-	}
-
-	function timerecord_menu( $p_event, $p_bug_id ) {
-		if( TimeTracking\user_can_view_bug_id( $p_bug_id ) ) {
-			$t_href = '#timerecord';
-			return array( plugin_lang_get( 'timerecord_menu' ) => $t_href );
-		}
-		else {
-			return array ();
-		}
-	}
-
-	function showreport_menu() {
-		return array(
-			array(
-				'title' => plugin_lang_get( 'title' ),
-				'access_level' => plugin_config_get( 'reporting_threshold' ),
-				'url' => plugin_page( 'report_page' ),
-				'icon' => 'fa-random'
-			)
-		);
-	}
+    function register() {
```

Comment:
Please follow Mantis general coding guidelines. Especially:
`Indentations: use TABS with a size of 4 (not spaces)`
Your commit has marked all the file as diff because it has turned tabs into spaces.